packages/build/src/core/normalize_flags.js

```diff
@@ -26,26 +33,31 @@ const normalizeFlags = function (flags, logs) {
 }
 
 // Default values of CLI flags
-const getDefaultFlags = function ({ env: envOpt = {}, mode = REQUIRE_MODE }) {
-  const combinedEnv = { ...env, ...envOpt }
+const getDefaultFlags = function ({ env: envOpt = {} }, combinedEnv) {
   return {
     env: envOpt,
     nodePath: execPath,
     token: combinedEnv.NETLIFY_AUTH_TOKEN,
     mode: REQUIRE_MODE,
     offline: false,
+    telemetry: false,
     functionsDistDir: DEFAULT_FUNCTIONS_DIST,
     deployId: combinedEnv.DEPLOY_ID,
     debug: Boolean(combinedEnv.NETLIFY_BUILD_DEBUG),
     bugsnagKey: combinedEnv.BUGSNAG_KEY,
-    telemetry: !combinedEnv.BUILD_TELEMETRY_DISABLED && mode !== REQUIRE_MODE,
     sendStatus: false,
     testOpts: {},
     featureFlags: DEFAULT_FEATURE_FLAGS,
     statsd: { port: DEFAULT_STATSD_PORT },
   }
 }
 
+// Compute the telemetry flag, it's disabled by default and we want to always disable it
+// if BUILD_TELEMETRY_DISABLED is passed
+const computeTelemetry = function (flags, envOpts) {
+  return envOpts.BUILD_TELEMETRY_DISABLED ? { telemetry: false } : { telemetry: flags.telemetry }
```

Comment:
[sand] Would it be simpler to remove `telemetry` from `getDefaultFlags()` and make this function return `flags.telemetry && !envOpts.BUILD_TELEMETRY_DISABLED` instead?